t/23-amqp.t

```diff
@@ -189,4 +190,60 @@ subtest 'create parent group comment' => sub {
     is($json->{parent_group_id}, 2000,  'parent group id');
 };
 
+# Now let's unmock publish_amqp so we can test it...
+$plugin_mock->unmock('publish_amqp');
+%published = ();
+# ...but we'll mock the thing it calls.
+my $publisher_mock = Test::MockModule->new('Mojo::RabbitMQ::Client::Publisher');
+$publisher_mock->mock(
+    publish_p => sub {
+        # copied from upstream git master as of 2019-07-24
+        my $self    = shift;
+        my $body    = shift;
+        my $headers = {};
+        my %args    = ();
+
+        if (ref($_[0]) eq 'HASH') {
+            $headers = shift;
+        }
+        if (@_) {
+            %args = (@_);
+        }
+        # end copying
+        $published{body}    = $body;
+        $published{headers} = $headers;
+        $published{args}    = \%args;
+        # we need to return a Promise or stuff breaks
+        my $client_promise = Mojo::Promise->new();
+        return $client_promise;
+    });
+
+# we need an instance of the plugin now. I can't find a documented
+# way to access the one that's already loaded...
+my $amqp = OpenQA::WebAPI::Plugin::AMQP->new;
+$amqp->register($app);
```

Comment:
You can't, it seems like a more correct solution would be to expose `publish_amqp` as a helper to the application.

Reply:
I mean, this works...so is it really necessary to do that just for the sake of the tests? If I'm going to do that we may as well make them both utility functions and then I can write a fedora AMQP plugin that isn't a subclass of the upstream one, but I kinda liked the subclass idea...